# HG changeset patch
# User welpy-cw <h.w.forms@arcor.de>
# Date 1706555225 -3600
# Node ID 096cfd237f7030e4ce58fc8146abb4dfd2abf065
# Parent  800a6e8769e650bba49c130920e7d9172a963a75
Bug 1873282 - Fix setting references when parsing IMAP mailbox. r=BenC

Clear the references information in nsIMsgDBHdrs for messages without any References or
In-Reply-To headers.

Differential Revision: https://phabricator.services.mozilla.com/D199514

diff --git a/mailnews/local/src/nsParseMailbox.cpp b/mailnews/local/src/nsParseMailbox.cpp
--- a/mailnews/local/src/nsParseMailbox.cpp
+++ b/mailnews/local/src/nsParseMailbox.cpp
@@ -1338,18 +1338,21 @@ nsresult nsParseMailMessageState::Finali
           }
         }
 
         if (account_key != nullptr)
           m_newMsgHdr->SetAccountKey(account_key->value);
         // use in-reply-to header as references, if there's no references header
         if (references != nullptr) {
           m_newMsgHdr->SetReferences(nsDependentCString(references->value));
-        } else if (inReplyTo != nullptr)
+        } else if (inReplyTo != nullptr) {
           m_newMsgHdr->SetReferences(nsDependentCString(inReplyTo->value));
+        } else {
+          m_newMsgHdr->SetReferences(EmptyCString());
+        }
 
         // 'Received' should be as reliable an indicator of the receipt
         // date+time as possible, whilst always giving something *from
         // the message*.  It won't use PR_Now() under any circumstance.
         // Therefore, the fall-thru order for 'Received' is:
         // Received: -> Delivery-date: -> date
         // 'Date' uses:
         // date -> 'Received' -> PR_Now()
